src/xpath.js

```diff
@@ -5,8 +5,48 @@ define([
     xpath,
     makeXPathModels
 ) {
-    var models = makeXPathModels();
-    window.xpathmodels = models;  // the xpath module uses this global
-    xpath.models = models;
-    return xpath;
+    var hashtagToXPath = {},
+        xpathmodels = makeXPathModels({
+        isValidNamespace: function (namespace) {
+            return namespace === 'form';
+        },
+        hashtagToXPath: function (hashtagExpr) {
+            if (hashtagToXPath[hashtagExpr]) {
+                return hashtagToXPath[hashtagExpr];
+            }
+            return hashtagExpr;
+        },
+        toHashtag: function (xpath_) {
+            function toHashtag(xpathExpr) {
+                for (var key in hashtagToXPath) {
+                    if (hashtagToXPath.hasOwnProperty(key)) {
+                        if (hashtagToXPath[key] === xpathExpr)
+                            return key;
+                    }
+                }
+                return null;
+            }
+
+            if (xpath_ instanceof xpathmodels.HashtagExpr) {
+                return xpath_.toHashtag();
+            }
+            return toHashtag(xpath_.toXPath());
+        }
+    });
+
+    window.xpathmodels = xpathmodels;
+
+    return {
+        parser: xpath,
+        models: xpathmodels,
+        setHashtagToXPathDict: function (translationDict) {
```

Comment:
I originally included these three methods in logic, but tried to stick to a process of use logic manager only for when you need to keep track of the particular references and use xpath for normalizing/translating to hashtags